python/cuml/tests/test_spectral_embedding.py

```diff
@@ -66,51 +68,127 @@ def load_digits_dataset(n_samples=None):
     return digits.data
 
 
-# Dataset configurations: (dataset_loader, dataset_name, n_samples, min_trustworthiness)
-dataset_configs = [
-    (generate_s_curve, 1500, 0.8),
-    (generate_s_curve, 2000, 0.8),
-    (generate_swiss_roll, 2000, 0.8),
-    (generate_swiss_roll, 3000, 0.8),
-    (generate_mnist_like_dataset, 5000, 0.8),
-    (load_digits_dataset, None, 0.8),
-]
-
-
 @pytest.mark.parametrize(
-    "dataset_loader,n_samples,min_trustworthiness",
-    dataset_configs,
+    "affinity,graph_type",
+    [
+        ("nearest_neighbors", None),  # Use built-in nearest_neighbors affinity
+        ("precomputed", "binary_knn"),  # Precomputed binary k-NN graph
+        ("precomputed", "fuzzy_knn"),  # Precomputed fuzzy k-NN graph from UMAP
+    ],
```

Comment:
Could we also add `("precomputed", "regular_knn")` with `mode="distance"` to check that it is as good as `("nearest_neighbors", None)`.